mono/metadata/object-internals.h

```diff
@@ -228,6 +228,7 @@ typedef struct {
 	MonoObject  *async_callback;
 	MonoObject  *execution_context;
 	MonoObject  *original_context;
+	guint32      gchandle;
```

Comment:
You can't just put this here. This represents a managed object that is allocated from the C# definition. When you use this field later to are writing to random memory.